changes discussed on 3/8/23 (#469)
moderndive/ModernDive_book: 07-sampling.Rmd

```

## A theoretical approach and the Central Limit Theorem

We now introduce a simple theoretical approach to understanding some important properties of the sample proportion. To do this we modify the bowl activity slightly. Instead of using a shovel to select all 25 balls at once, we randomly select one ball at a time. If the ball is red we call it a success and record a 1 (one); if it is not red we call it a failure and record a 0 (zero). Then, we return the ball to the bowl so the proportion of red balls in the bowl doesn't change. 

> This process is called a trial or a Bernoulli trial in honor of Jacob Bernoulli, a 17th-century mathematician who used it already back in those times. 

Reviewer: ```suggestion
This process is called a trial or (more formally) a Bernoulli trial in honor of Jacob Bernoulli, a 17th-century mathematician who used it already back in those times. 
```

Getting a sample of 25 balls is running 25 trials and obtaining 25 numbers, ones or zeros, representing whether we have observed red balls or not on each trial, respectively. 

> The average of these 25 numbers is the sample proportion. 

Reviewer: @avaldivi6 I think laying out an example here to show that the average of these 25 numbers will result in the sample proportion would be helpful.

Reviewer: You do introduce what this looks like with terminology just after this, but we've usually followed laying things out without the terminology and then assigning terminology to reduce cognitive load.

### Random Variables

> It is useful to represent a trial as a random variable that takes the value of 1 if the ball is red or the value of 0 if the ball is not red. 

Reviewer: @avaldivi6 Maybe explain a little more both of the words "random" and "variable" here from a novice perspective? We don't know whether a given ball chosen will be red or not red so the event in question is random. It will likely vary from one selection to another. Something along those lines.

Reviewer: @avaldivi6 It looks like this wasn't updated. Would you like to discuss further on our next call?

Author: @ismayc Yes, let's talk about this

We denote the first trial $X_1$, the second trial $X_2$, and so on. 
The average of these 25 trials, before they are carried out, can also be expressed as a random variable: $$\bar X = \frac{X_1+X_2+\dots+X_{25}}{25}.$$

> Note that $\bar X$ is a random variable before the trials are run, so $\bar X$ represents all the possible sample proportions that we could obtain. After we run the first trial, if the ball observed is red, we assign $X_1 = 1$, and if it is not red $X_1=0$. Once all 25 trials have been performed, we get a list of 25 ones or zeros. As an illustration, one possible outcome could be: $$\{0,0,0,1,0,1,0,1,0,0,1,0,0,1,1,0,0,0,1,1,0,1,0,0,0,1 \}$$  The observed value of $\bar X$ will be 

Reviewer: ```suggestion
Note that $\bar X$ is a random variable before the trials are run, so $\bar X$ represents all the possible sample proportions that we could obtain. After we run the first trial, if the ball observed is red, we assign $X_1 = 1$, and if it is not red $X_1=0$. Once all 25 trials have been performed, we get a list of 25 ones or zeros. As an illustration, one possible outcome could be: $$\{0,0,0,1,0,1,0,1,0,0,1,0,0,1,1,0,0,0,1,1,0,1,0,0,0,1 \}$$  For this one possible outcome, the observed value of $\bar X$ will be 
```

$$\bar X = \frac{0+0+0+1+\dots+0+1}{25} = \frac{10}{25}=0.4.$$ 
which is both the sample mean of those trial runs and the sample proportion of red balls in the observed sample. 

> Sample proportions are sample means!

Reviewer: Hmm, maybe this is OK. But I think one additional example walking through another possible outcome before introducing the terminology would be helpful.

### The distribution of $\bar X$

The sampling distribution of the sample proportion (or sample mean) is now represented by the distribution of $\bar X$. Think of this distribution as how often each of the values that $\bar X$ could take may end up happening. 

A very useful way to learn about this distribution is using simulations. 

> Simulations seldom provide the exact structure of the distribution, because we would need an infinite number of samples, but a large number of replications often produce a really good approximation of the distribution and we can use it to study the distribution's characteristics. We now repeat our experiment 10,000 times. 

Reviewer: ```suggestion
Simulations seldom provide the exact structure of the distribution, because we would need an infinite number of samples to get the exact layout. A large number of replications can often produce a really good approximation of the exact distribution, and we can use it to study the distribution's characteristics. We now repeat our experiment 10,000 times. 
```

```{r}
simul_X_bar_25 <- bowl |>
  rep_sample_n(size = 25, replace = TRUE, reps = 10000) |>
  summarize(prop_red = mean(color == "red"))
```

> The object `simul_X_bar` contains 10,000 sample proportions, each replicate was obtained following the experiment described earlier, so each sample proportion is a possible value of $\bar X$. 

Reviewer: ```suggestion
The object `simul_X_bar_25` contains 10,000 sample proportions of our process of drawing 25 balls; each replicate was obtained following the experiment described earlier. Thus, each sample proportion is a possible value of $\bar X$. 
```

### The mean and standard deviation of $\bar X$

> Based on probability theory, the mean of $\bar X$ should be equal to the population proportion of red balls in the bowl, that is $p = 900/2400 = 0.375$, and the standard deviation of $\bar X$, also called the standard error, is given by the formula: $$SE_{\bar X} = \sqrt{\frac{p(1-p)}{n}} = \sqrt{\frac{0.375\cdot(1-0.375)}{25}} = 0.0968$$

Reviewer: I think a reference back to what we've seen previously (before the theory) would be helpful here and then formalize it.

  mutate(n = 100)

# n = 400
if (!file.exists("rds/virtual_samples_400r.rds")) {
  virtual_samples_400r <- bowl |>
    rep_sample_n(size = 400, replace = TRUE, reps = 10000)
  write_rds(virtual_samples_400r, "rds/virtual_samples_400r.rds")
} else {
  virtual_samples_400r <- read_rds("rds/virtual_samples_400r.rds")
}
virtual_prop_red_400r <- virtual_samples_400r |>
  group_by(replicate) |>
  summarize(red = sum(color == "red")) |>
  mutate(prop_red = red / 400) |>
  mutate(n = 400)

virtual_prop <- bind_rows(
  virtual_prop_red_25r,
  virtual_prop_red_100r,
  virtual_prop_red_400r
)

comparing_sampling_distributions <- ggplot(virtual_prop, aes(x = prop_red)) +
  geom_histogram(aes(y = ..density..), binwidth = 0.04, boundary = 0.4, color = "white") +
  labs(
    x = "The distribution of sample proportions for red balls",
    title = "Comparing histograms of sample proportions for three different sample sizes"
  ) +
  facet_wrap(~n)

if (is_latex_output()) {
  comparing_sampling_distributions +
    theme(
      strip.text = element_text(colour = "black"),
      strip.background = element_rect(fill = "grey93")
    )
} else {
  comparing_sampling_distributions
}
```

 

> Moreover, the shape of the histograms provides a really good approximation of the shape of the distribution of the sample proportion. If we were to draw a single smooth curve that connects the top of each bar on a histogram with each adjacent bar, the curves for each histogram would look similar to the ones presented in Figures  \@ref(fig:sample-proportion-25-with-normal-pdf), \@ref(fig:sample-proportion-100-with-normal-pdf), and \@ref(fig:sample-proportion-400-with-normal-pdf). However, the curves were not drawn using simulated data; these bell-shaped curves are curves for the normal distribution with mean equal to $p=0.375$ and standard deviation equal to $\sqrt{\frac{p(1-p)}{n}}$ where $n$ changes for each histogram. This is a fascinating result but it is not unexpected. It relies on an application of one of the most important results in Statistics: the Central Limit Theorem (CLT).

Reviewer: ```suggestion
Moreover, the shape of the histograms provides an excellent approximation of the shape of the distribution of the sample proportion. If we were to draw a single smooth curve that connects the top of each bar on the histogram with each adjacent bar, the curves for each histogram would look similar to the ones presented in Figures  \@ref(fig:sample-proportion-25-with-normal-pdf), \@ref(fig:sample-proportion-100-with-normal-pdf), and \@ref(fig:sample-proportion-400-with-normal-pdf). However, the curves were not drawn using simulated data; these bell-shaped curves are curves for the normal distribution with mean equal to $p=0.375$ and standard deviation equal to $\sqrt{\frac{p(1-p)}{n}}$ where $n$ changes for each histogram. This is a fascinating result but it is not unexpected. It relies on an application of one of the most important results in Statistics: the Central Limit Theorem (CLT).
```

Reviewer: Very nice! This is a cleaner introduction to CLT.

The CLT states that when the sample size, $n$, tends to infinity, the distribution of $bar X$ tends to the normal distribution (with the appropriate mean and standard deviation). 
Moreover, it does not depend on the population distribution; the population can be a bowl with red and white balls, the number of daily births in the US, or anything else. As long as we work with the sample mean (or sample proportion), $\bar X$, the CLT will provide the same solution.

The observant reader may have noticed that, while this result seems really fascinating, in practice we cannot take samples of size equal to infinity. 
What makes the CLT even more relevant, in practice, is that even for small sample sizes, the distribution of $bar X$ approximates normality very quickly. As you can see in Figure  \@ref(fig:sample-proportion-25-with-normal-pdf), even for a sample of size $n=25$ the distribution of sample proportions already appears to follow a normal distribution.

```{r sample-proportion-25-with-normal-pdf, echo=FALSE, fig.height=3, fig.cap="Histogram of the distribution of the sample proportion and the normal curve.", purl=FALSE}

> n = 25

Reviewer: ```suggestion
n <- 25
```

> p=9/24

Reviewer: ```suggestion
p <- 9/24
```

Reviewer: Can you change the other `=` to `<-` below in this block too?

> sd.p = sqrt(p*(1-p)/n)

Reviewer: ```suggestion
sd.p <- sqrt(p*(1-p)/n)
```

Reviewer: Can you change this to `sd_p` instead of `sd.p`?

if (!file.exists("rds/virtual_samples_25r.rds")) {
  virtual_samples_25r <- bowl |>
    rep_sample_n(size = 25, replace = TRUE, reps = 10000)
  write_rds(virtual_samples_25, "rds/virtual_samples_25r.rds")
} else {
  virtual_samples_25r<- read_rds("rds/virtual_samples_25r.rds")
}

ggplot(virtual_prop_red_25r, aes(x = prop_red)) +

>   geom_histogram(aes(y=..density..), binwidth = 0.04, color = "white") + 

Reviewer: ```suggestion
  geom_histogram(aes(y = ..density..), binwidth = 0.04, color = "white") + 
```

  geom_histogram(aes(y=..density..), binwidth = 0.01, color = "white") + 
  stat_function(fun = dnorm,  args = list(mean = p, sd = sd.p)) + xlim(0,0.75) + ylim(0,18)+
  labs(
    x = "Sample proportions for red balls from samples of size 100",
    title = "Histogram for the sample proportions"
  )
```

```{r sample-proportion-400-with-normal-pdf, echo=FALSE, fig.height=3, fig.cap="Histogram of the distribution of the sample proportion and the normal curve.", purl=FALSE}
n = 400
p=9/24
sd.p = sqrt(p*(1-p)/n)

if (!file.exists("rds/virtual_samples_400r.rds")) {
  virtual_samples_400r <- bowl |>
    rep_sample_n(size = 400, replace = TRUE, reps = 10000)
  write_rds(virtual_samples_400r, "rds/virtual_samples_400r.rds")
} else {
  virtual_samples_400r<- read_rds("rds/virtual_samples_400r.rds")
}

ggplot(virtual_prop_red_400r, aes(x = prop_red)) +
  geom_histogram(aes(y=..density..), binwidth = 0.0025, color = "white") + 
  stat_function(fun = dnorm,  args = list(mean = p, sd = sd.p)) + xlim(0,0.75) + ylim(0,18)+
  labs(
    x = "Sample proportions for red balls from samples of size 400",
    title = "Histogram for the sample proportions"
  )
```

> The same happened in Figures \@ref(fig:sample-proportion-100-with-normal-pdf) and \@ref(fig:sample-proportion-400-with-normal-pdf) when sample proportions of larger sample sizes are obtained. The range of the $x-$ and $y-$axis on all these figures has been kept constant for appropriate comparisons. The figures represent density histograms where the area of each bar represents the percentage or proportion of observations for the corresponding class and the total area of each histogram is 1 (or 100%). Observe also that all the curves follow the bell-shaped form of the normal curve but the spread is greater the smaller the sample size used. The spread observed are consistent with the value of the standard deviation for $\bar X$ (or standard error) that we obtained earlier for each case.

Reviewer: ```suggestion
The same happened in Figures \@ref(fig:sample-proportion-100-with-normal-pdf) and \@ref(fig:sample-proportion-400-with-normal-pdf) when sample proportions of larger sample sizes are obtained. The range of the $x-$ and $y-$axis on all these figures has been kept constant for appropriate comparisons. The figures represent density histograms where the area of each bar represents the percentage or proportion of observations for the corresponding class and the total area of each histogram is 1 (or 100%). Observe also that all the curves follow the bell-shaped form of the normal curve but the spread is greater the smaller the sample size used. The spreads observed are consistent with the value of the standard deviation for $\bar X$ (or standard error) that we obtained earlier for each case.
```
